sdk/core/core-rest-pipeline/test/multipartPolicy.spec.ts

```diff
@@ -0,0 +1,336 @@
+        "fully specified content-type header should be preserved"
+      );
+    });
+
+    it("sets the boundary when it is specified in the multipart request body", async function () {
+      const request = await performRequest({
+        headers: createHttpHeaders({
+          "content-type": "multipart/alternative",
+        }),
+        body: {
+          boundary: "blah",
+          parts: [],
+        },
+      });
+
+      assert.equal(
+        request.headers.get("content-type"),
+        "multipart/alternative; boundary=blah",
+        "boundary was not added"
+      );
+    });
+  });
+
+  describe("multipart request body", function () {
+    it("request with no parts matches spec", async function () {
+      const request = await performRequest({
+        body: {
+          boundary: "blah",
+          parts: [],
+        },
+      });
+
+      const expectedBody = stringToUint8Array("--blah--\r\n\r\n", "utf-8");
+      await assertBodyMatches(request.body, expectedBody);
+    });
+
+    describe("boundary", function () {
+      it("is present with multiple parts", async function () {
+        const request = await performRequest({
+          body: {
+            boundary: "blah",
+            parts: [
+              {
+                body: stringToUint8Array("part1", "utf-8"),
+                headers: createHttpHeaders(),
+              },
+              {
+                body: stringToUint8Array("part2", "utf-8"),
+                headers: createHttpHeaders(),
+              },
+            ],
+          },
+        });
+
+        const expectedBody = stringToUint8Array(
+          "--blah\r\n\r\npart1\r\n--blah\r\n\r\npart2\r\n--blah--\r\n\r\n",
+          "utf-8"
+        );
+        await assertBodyMatches(request.body, expectedBody);
+        assert.equal(
+          request.headers.get("Content-Length"),
+          expectedBody.byteLength.toString(),
+          "Expected Content-Length header to equal length of body"
+        );
+      });
+
+      it("supports Uint8Array body", async function () {
+        const request = await performRequest({
+          body: {
+            boundary: "blah",
+            parts: [
+              {
+                body: stringToUint8Array("part", "utf-8"),
+                headers: createHttpHeaders(),
+              },
+            ],
+          },
+        });
+
+        const expectedBody = stringToUint8Array("--blah\r\n\r\npart\r\n--blah--\r\n\r\n", "utf-8");
+        await assertBodyMatches(request.body, expectedBody);
+        assert.equal(
+          request.headers.get("Content-Length"),
+          expectedBody.byteLength.toString(),
+          "Expected Content-Length header to equal length of body"
+        );
+      });
+
+      it("supports Node ReadableStream body", async function () {
+        if (!isNode) {
+          this.skip();
+        }
+
+        const body = Readable.from(Buffer.from("part1", "utf-8"));
+
+        const request = await performRequest({
```

Comment:
again perhaps a test with more exotic body contents just to make sure the textencoder is working properly

Reply:
multipartPolicy only takes binary content -- stream or Uint8Array. It's the caller's responsibility to encode their text as they see fit so I don't think we need to test this here (good call on doing it in formDataPolicy though)

Reply:
i see we do have `string` in RequestBodyType -- maybe we should add that to `BodyPart` too?